discord/onboarding.py

```diff
@@ -0,0 +1,347 @@
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from __future__ import annotations
+
+import os
+from typing import TYPE_CHECKING, Iterable, Optional, Set, List, Union
+
+from discord.utils import MISSING
+
+from . import utils
+from .mixins import Hashable
+from .enums import OnboardingMode, OnboardingPromptType, try_enum
+from .utils import cached_slot_property, MISSING
+
+__all__ = (
+    'Onboarding',
+    'PartialOnboardingPrompt',
+    'OnboardingPrompt',
+    'OnboardingPromptOption',
+    'PartialOnboardingPromptOption',
+)
+
+
+if TYPE_CHECKING:
+    from .abc import GuildChannel
+    from .emoji import Emoji
+    from .guild import Guild
+    from .partial_emoji import PartialEmoji
+    from .role import Role
+    from .threads import Thread
+    from .types.onboarding import (
+        Prompt as PromptPayload,
+        PromptOption as PromptOptionPayload,
+        Onboarding as OnboardingPayload,
+    )
+    from .state import ConnectionState
+
+
+class PartialOnboardingPromptOption:
+    """Represents a partial onboarding prompt option, these are used in the creation
+    of an :class:`OnboardingPrompt` via :meth:`Guild.edit_onboarding`.
+
+    .. versionadded:: 2.4
+
+    Attributes
+    -----------
+    title: :class:`str`
+        The title of this prompt option.
+    description: Optional[:class:`str`]
+        The description of this prompt option.
+    emoji: Optional[Union[:class:`Emoji`, :class:`PartialEmoji`, :class:`str`]]
+        The emoji tied to this option. May be a custom emoji, or a unicode emoji.
+    channel_ids: Set[:class:`int`]
+        The IDs of the channels that will be made visible if this option is selected.
+    role_ids: Set[:class:`int`]
+        The IDs of the roles given to the user if this option is selected.
+    """
+
+    __slots__ = (
+        'title',
+        'emoji',
+        'description',
+        'channel_ids',
+        'role_ids',
+    )
+
+    def __init__(
+        self,
+        title: str,
+        emoji: Union[Emoji, PartialEmoji, str],
+        description: Optional[str] = None,
+        channel_ids: Iterable[int] = MISSING,
+        role_ids: Iterable[int] = MISSING,
+    ) -> None:
+        self.title: str = title
+        self.description: Optional[str] = description
+        self.emoji: Union[PartialEmoji, Emoji, str] = emoji
+        self.channel_ids: Set[int] = set(channel_ids or [])
+        self.role_ids: Set[int] = set(role_ids or [])
+
+    def to_dict(self, *, id: int = MISSING) -> PromptOptionPayload:
+        if isinstance(self.emoji, str):
+            emoji_payload = {"emoji_name": self.emoji}
+        else:
+            emoji_payload = {
+                "emoji_id": self.emoji.id,
+                "emoji_name": self.emoji.name,
+                "emoji_animated": self.emoji.animated,
+            }
+
+        return {
+            'id': id or os.urandom(16).hex(),
```

Comment:
Wouldn't this be always ``MISSING`` when no value is passed🤔? Shouldn't this be a ``id if id is not MISSING else os.urandom(16).hex()``?